cms/static/js/views/pages/container.js

```diff
@@ -322,6 +322,22 @@ define(['jquery', 'underscore', 'backbone', 'gettext', 'js/views/pages/base_page
             },
 
             onNewXBlock: function(xblockElement, scrollOffset, is_duplicate, data) {
+                var useNewTextEditor = this.$('.xblock-header-primary').attr("use-new-editor-text"),
+                useNewVideoEditor = this.$('.xblock-header-primary').attr("use-new-editor-video"),
+                useNewProblemEditor = this.$('.xblock-header-primary').attr("use-new-editor-problem");
+
+                //find the block type in the locator
+                var matchBlockTypeFromLocator = /(?<=\@)(.*?)(?=\+)/g;
```

Comment:
yes this is a janky solution, open to others. Why we do this is because on create, we merely create a locator, and send that off to the api. We don't have that data in Studio till we cal refresh (which we don't want to do be cause we want to navigate away before refresh.)